### PR TITLE
Rust 1.73

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

Upgrade to Rust 1.73

## Test Plan

`cargo test`

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #8007** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8007" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/8007?utm_source=stack-comment).

File: crates/ruff_notebook/src/notebook.rs [hunk 1]
```diff
@@ -149,7 +149,7 @@ impl Notebook {
     {
         let trailing_newline = reader.seek(SeekFrom::End(-1)).is_ok_and(|_| {
             let mut buf = [0; 1];
-            reader.read_exact(&mut buf).is_ok_and(|_| buf[0] == b'\n')
+            reader.read_exact(&mut buf).is_ok_and(|()| buf[0] == b'\n')
```
Comment (konstin) [inline crates/ruff_notebook/src/notebook.rs:152]: Not convinced by that clippy change tbh (the other ones are good though)

**Review by konstin — APPROVED** [review]

File: crates/ruff_notebook/src/notebook.rs (hunk 1 above)
Comment (MichaReiser) [reply crates/ruff_notebook/src/notebook.rs:152]: I don't mind disabling that lint... Here's the documentation of the rule and why https://rust-lang.github.io/rust-clippy/master/index.html#/ignored_unit_patterns

I was like, meh but then it's rare and not too bad

**Review by dhruvmanila — APPROVED** [review]

File: crates/ruff_notebook/src/notebook.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/ruff_notebook/src/notebook.rs:152]: I'm fine with it, don't feel strongly.

**Comment by charliermarsh** [comment]
I'd been holding off on this as Homebrew doesn't yet support it (conda-forge does): https://github.com/Homebrew/homebrew-core/pull/149754

**Comment by MichaReiser** [comment]
> I'd been holding off on this as Homebrew doesn't yet support it (conda-forge does): [Homebrew/homebrew-core#149754](https://github.com/Homebrew/homebrew-core/pull/149754)

This should be fine as far as I understand because I only bumped our toolchain version but not the minimal required Rust version (Cargo.toml)

**Comment by MichaReiser** [comment]
Just tested `brew install ruff --head` and it compiles fine :)

**Comment by chenrui333** [comment]
FYI, rust 1.74.0 got shipped in the homebrew side now, https://github.com/Homebrew/homebrew-core/pull/154526

**Comment by charliermarsh** [comment]
Thank you @chenrui333!

**Pull request merged**